optaplanner-examples/src/main/java/org/optaplanner/examples/examination/solver/score/ExaminationConstraintProvider.java

```diff
@@ -65,12 +83,10 @@ protected Constraint periodDurationTooShort(ConstraintFactory constraintFactory)
     }
 
     protected Constraint roomCapacityTooSmall(ConstraintFactory constraintFactory) {
-        return constraintFactory.from(Room.class)
-                .join(Exam.class, equal(Function.identity(), Exam::getRoom))
-                .join(Period.class, equal((room, exam) -> exam.getPeriod(), Function.identity()))
-                .groupBy((room, exam, period) -> room,
-                        (room, exam, period) -> period,
-                        ConstraintCollectors.sum((period, exam, room) -> exam.getTopicStudentSize()))
+        return constraintFactory.from(Exam.class)
+                .filter(exam -> exam.getRoom() != null)
+                .filter(exam -> exam.getPeriod() != null)
```

Comment:
a from() guarantees that all genuine planning variables aren't null.
So the room filter is definitely obsolete, because room is a planning variable on Exam.

For period its more complex. It's a genuine planning variable on LeadingExam, and a shadow variable on FollowingExam - and nothing on Exam. So I think it doesn't get filtered by default, because we're doing "from" Exam.
I would recommend adding a comment above that filter, to signal why were're filtering it.